shared/ghutil/fakeghutil/fakeghutil.go

```diff
@@ -185,6 +189,134 @@ func (fgc *FakeGithubClient) RemoveLabelForIssue(org, repo string, issueNumber i
 	return nil
 }
 
+// ListPullRequests lists pull requests within given repo, filters by head user and branch name if
+// provided as "user:ref-name", and by base name if provided, i.e. "master"
+func (fgc *FakeGithubClient) ListPullRequests(org, repo, head, base string) ([]*github.PullRequest, error) {
+	var res []*github.PullRequest
+	PRs, ok := fgc.PullRequests[repo]
+	if !ok {
+		return res, fmt.Errorf("repo %s not exist", repo)
+	}
+	for _, PR := range PRs {
+		// Filter with consistent logic of CreatePullRequest function below
+		if ("" == head || *PR.Head.Label == head) &&
+			("" == base || *PR.Base.Ref == base) {
+			res = append(res, PR)
+		}
+	}
+	return res, nil
+}
+
+// ListCommits lists commits from a pull request
+func (fgc *FakeGithubClient) ListCommits(org, repo string, ID int) ([]*github.RepositoryCommit, error) {
+	commits, ok := fgc.PRCommits[ID]
```

Comment:
why no just `return fgc.PRCommits[ID]`

The caller will get an empty array, if not present

Reply:
ok is a bool, but the function needs error, so it would not work